### ISSUE TITLE
Suggest here-string rather than `echo` + a pipeline

### ISSUE THREAD

**Issue by l0b0** [koalaman/shellcheck#2593]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet that shows the problem:

```sh
#!/usr/bin/env bash
echo value | foo
```

Suggested code:

```sh
#!/usr/bin/env bash
foo <<< value
```

#### Here's what shellcheck currently says:

> No issues detected!

#### Here's what I wanted or expected to see:

> Use a here document instead of `echo` + a pipeline.

[Rationale](https://paperless.blog/bash-lints-beyond-shellcheck): This is analogous to the [useless cat](https://github.com/koalaman/shellcheck/wiki/SC2002) lint.

Exceptions: If you’re a strong proponent of left-to-right flow of information you can also use `<<< value foo`, but that doesn't seem worth the confusion.
